akka-remote/src/main/scala/akka/remote/artery/tcp/ssl/X509Readers.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.remote.artery.tcp.ssl
+
+import java.security.cert.X509Certificate
+import java.util
+
+import akka.annotation.InternalApi
+import javax.naming.ldap.LdapName
+import akka.util.ccompat.JavaConverters._
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] object X509Readers {
+
+  def getAllSubjectNames(cert: X509Certificate): Set[String] = {
+    val maybeCommonName =
+      new LdapName(cert.getSubjectX500Principal.getName).getRdns.asScala.collectFirst {
+        case attr if attr.getType.equalsIgnoreCase("CN") =>
+          attr.getValue.toString
+      }
```

Comment:
May want to think about error handling here - failing/throwing is probably the right thing to do, but might want to wrap in an exception that gives more useful error info in what was trying to be achieved.

Reply:
This is internal and only used in the `PeerSubjectVerifier` code. The only reason why `X509readers` is extracted is to unit test it.

I've been trying to cause a failure in the `X509readers` code with several certificates. All of the inputs I used in the tests are valid x.509 certificates. I've used different values for the CN in the `DName` and the `DNS` field in the SAN extension: no SAN, SAN without DNS, DName without CN, duplicate entries,... Even when there's no subject at all, `getAllSubjectNames()` returns the expected `Set()`. I haven't pushed this tests yet to not extend this PR even more.

The other possible failure I can think of could be caused if the file provided by the user is not a valid `X509Certificate`, in which case this code is unreachable.

If you have any other cases in mind, I think we can make the improvement on a separate PR.

Reply:
pushed these tests in https://github.com/ignasi35/akka/commit/5da4fc6e4b514f6cce03d7014714b15052b98a1c